File: c/src/catalogo.c
// src/catalogo.c


#include <stdio.h>
#include "catalogo.h"

void inicializarCatalogo(Catalogo *c) {
    c->total = 0;
}

void adicionarLivro(Catalogo *c, Livro livro) {
    if (c->total < MAX_LIVROS) {
        c->livros[c->total] = livro;
        c->total++;
    } else {
        printf("Catálogo cheio!\n");
    }
}

void listarCatalogo(Catalogo c) {
    printf("\n=== CATÁLOGO DE LIVROS ===\n\n");

    for (int i = 0; i < c.total; i++) {
        printf("Livro %d\n", i + 1);
        mostrarLivro(c.livros[i]);
        printf("\n");
    }
}

File: java/src/data/Catalogo.java
// src/data/Catalogo.java


package data;

import java.util.ArrayList;
import models.LivroBiblioteca;

public class Catalogo {

    private ArrayList<LivroBiblioteca> livros;

    // Construtor
    public Catalogo() {
        livros = new ArrayList<>();
    }

    // Adicionar livro ao catálogo
    public void adicionarLivro(LivroBiblioteca livro) {
        livros.add(livro);
    }

    // Obter todos os livros
    public ArrayList<LivroBiblioteca> getLivros() {
        return livros;
    }

    // Obter um livro pelo índice (posição)
    public LivroBiblioteca getLivro(int indice) {
        return livros.get(indice);
    }

    // Número total de livros
    public int totalLivros() {
        return livros.size();
    }
}
